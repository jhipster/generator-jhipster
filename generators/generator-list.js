/**
 * Copyright 2013-2022 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project, see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const Generators = {
  GENERATOR_ADD: 'add',
  GENERATOR_APP: 'app',
  GENERATOR_AWS: 'aws',
  GENERATOR_AZURE_APP_SERVICE: 'azure-app-service',
  GENERATOR_AZURE_SPRING_CLOUD: 'azure-spring-cloud',
  GENERATOR_BASE: 'base',
  GENERATOR_BOOTSTRAP: 'bootstrap',
  GENERATOR_BOOTSTRAP_APPLICATION: 'bootstrap-application',
  GENERATOR_CI_CD: 'ci-cd',
  /* @deprecated */
  GENERATOR_CICD: 'ci-cd',
  GENERATOR_CLIENT: 'client',
  GENERATOR_CLOUDFOUNDRY: 'cloudfoundry',
  GENERATOR_COMMON: 'common',
  GENERATOR_CYPRESS: 'cypress',
  GENERATOR_DATABASE_CHANGELOG: 'database-changelog',
  GENERATOR_DATABASE_CHANGELOG_LIQUIBASE: 'database-changelog-liquibase',
  GENERATOR_DOCKER_COMPOSE: 'docker-compose',
  GENERATOR_ENTITIES: 'entities',
  GENERATOR_ENTITIES_CLIENT: 'entities-client',
  GENERATOR_ENTITY: 'entity',
  GENERATOR_ENTITY_CLIENT: 'entity-client',
  GENERATOR_ENTITY_I_18_N: 'entity-i18n',
  /* @deprecated */
  GENERATOR_ENTITY_I18N: 'entity-i18n',
  GENERATOR_ENTITY_SERVER: 'entity-server',
  GENERATOR_EXPORT_JDL: 'export-jdl',
  GENERATOR_GAE: 'gae',
  GENERATOR_GENERATE_BLUEPRINT: 'generate-blueprint',
  GENERATOR_GRADLE: 'gradle',
  GENERATOR_HEROKU: 'heroku',
  GENERATOR_INFO: 'info',
  GENERATOR_INIT: 'init',
  GENERATOR_JAVA: 'java',
  GENERATOR_KUBERNETES: 'kubernetes',
  GENERATOR_KUBERNETES_HELM: 'kubernetes-helm',
  GENERATOR_KUBERNETES_KNATIVE: 'kubernetes-knative',
  GENERATOR_LANGUAGES: 'languages',
  GENERATOR_MAVEN: 'maven',
  GENERATOR_OPENAPI_CLIENT: 'openapi-client',
  GENERATOR_OPENSHIFT: 'openshift',
  GENERATOR_PAGE: 'page',
  GENERATOR_PROJECT_NAME: 'project-name',
  GENERATOR_SERVER: 'server',
  GENERATOR_SPRING_BOOT: 'spring-boot',
  GENERATOR_SPRING_CONTROLLER: 'spring-controller',
  GENERATOR_SPRING_SERVICE: 'spring-service',
  GENERATOR_UPGRADE: 'upgrade',
  GENERATOR_UPGRADE_CONFIG: 'upgrade-config',
  GENERATOR_WORKSPACES: 'workspaces',
};

module.exports = Generators;
