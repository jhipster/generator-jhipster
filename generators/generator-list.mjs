/**
 * Copyright 2013-2023 the original author or authors from the JHipster project.
 *
 * This file is part of the JHipster project; see https://www.jhipster.tech/
 * for more information.
 *
 * Licensed under the Apache License; Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing; software
 * distributed under the License is distributed on an "AS IS" BASIS;
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND; either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const GENERATOR_ANGULAR = 'angular';
export const GENERATOR_APP = 'app';
export const GENERATOR_AWS = 'aws';
export const GENERATOR_AZURE_APP_SERVICE = 'azure-app-service';
export const GENERATOR_AZURE_SPRING_CLOUD = 'azure-spring-cloud';
export const GENERATOR_BASE = 'base';
export const GENERATOR_BASE_APPLICATION = 'base-application';
export const GENERATOR_BASE_DOCKER = 'base-docker';
export const GENERATOR_BOOTSTRAP = 'bootstrap';
export const GENERATOR_BOOTSTRAP_APPLICATION = 'bootstrap-application';
export const GENERATOR_BOOTSTRAP_APPLICATION_BASE = 'bootstrap-application-base';
export const GENERATOR_BOOTSTRAP_APPLICATION_CLIENT = 'bootstrap-application-client';
export const GENERATOR_BOOTSTRAP_APPLICATION_SERVER = 'bootstrap-application-server';
export const GENERATOR_CASSANDRA = 'cassandra';
export const GENERATOR_CI_CD = 'ci-cd';
export const GENERATOR_CLIENT = 'client';
export const GENERATOR_CLOUDFOUNDRY = 'cloudfoundry';
export const GENERATOR_COMMON = 'common';
export const GENERATOR_COUCHBASE = 'couchbase';
export const GENERATOR_CYPRESS = 'cypress';
export const GENERATOR_DOCKER = 'docker';
export const GENERATOR_DOCKER_COMPOSE = 'docker-compose';
export const GENERATOR_ELASTICSEARCH = 'elasticsearch';
export const GENERATOR_ENTITIES = 'entities';
export const GENERATOR_ENTITY = 'entity';
export const GENERATOR_EXPORT_JDL = 'export-jdl';
export const GENERATOR_GAE = 'gae';
export const GENERATOR_GENERATE_BLUEPRINT = 'generate-blueprint';
export const GENERATOR_GIT = 'git';
export const GENERATOR_GRADLE = 'gradle';
export const GENERATOR_HEROKU = 'heroku';
export const GENERATOR_INFO = 'info';
export const GENERATOR_INIT = 'init';
export const GENERATOR_KAFKA = 'kafka';
export const GENERATOR_KUBERNETES = 'kubernetes';
export const GENERATOR_KUBERNETES_HELM = 'kubernetes-helm';
export const GENERATOR_KUBERNETES_KNATIVE = 'kubernetes-knative';
export const GENERATOR_LANGUAGES = 'languages';
export const GENERATOR_LIQUIBASE = 'liquibase';
export const GENERATOR_LIQUIBASE_CHANGELOGS = 'liquibase-changelogs';
export const GENERATOR_MAVEN = 'maven';
export const GENERATOR_MONGODB = 'mongodb';
export const GENERATOR_OPENAPI_CLIENT = 'openapi-client';
export const GENERATOR_OPENSHIFT = 'openshift';
export const GENERATOR_PAGE = 'page';
export const GENERATOR_PROJECT_NAME = 'project-name';
export const GENERATOR_REACT = 'react';
export const GENERATOR_SERVER = 'server';
export const GENERATOR_SPRING_CONTROLLER = 'spring-controller';
export const GENERATOR_SPRING_SERVICE = 'spring-service';
export const GENERATOR_UPGRADE = 'upgrade';
export const GENERATOR_VUE = 'vue';
export const GENERATOR_WORKSPACES = 'workspaces';
