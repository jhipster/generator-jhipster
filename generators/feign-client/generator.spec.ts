import { basename, dirname } from 'path';
import { fileURLToPath } from 'url';
import { before, it, describe, expect } from 'esmocha';
import { checkEnforcements, defaultHelpers as helpers, runResult } from '../../test/support/index.js';
import { shouldSupportFeatures, testBlueprintSupport } from '../../test/support/tests.js';
import Generator from '../server/index.js';

import { GENERATOR_FEIGN_CLIENT } from '../generator-list.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

const generator = basename(__dirname);

describe('generator - feign-client', () => {
  it('generator-list constant matches folder name', async () => {
    await expect((await import('../generator-list.js')).GENERATOR_FEIGN_CLIENT).toBe(generator);
  });
  shouldSupportFeatures(Generator);
  describe('blueprint support', () => testBlueprintSupport(generator));
  checkEnforcements({}, GENERATOR_FEIGN_CLIENT);

  describe('with jwt', () => {
    before(async () => {
      await helpers.runJHipster(GENERATOR_FEIGN_CLIENT).withJHipsterConfig({ authenticationType: 'jwt' });
    });

    it('should match generated files snapshot', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });

  describe('with oauth2', () => {
    before(async () => {
      await helpers.runJHipster(GENERATOR_FEIGN_CLIENT).withJHipsterConfig({ authenticationType: 'oauth2' });
    });

    it('should match generated files snapshot', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
