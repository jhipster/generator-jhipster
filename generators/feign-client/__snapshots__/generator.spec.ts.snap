// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator - feign-client with jwt should match generated files snapshot 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/UserFeignClientInterceptor.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/FeignConfiguration.java": {
    "stateCleared": "modified",
  },
}
`;

exports[`generator - feign-client with oauth2 should match generated files snapshot 1`] = `
{
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/AuthorizationHeaderUtil.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/AuthorizedFeignClient.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/OAuth2InterceptedFeignConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/OAuthIdpTokenResponseDTO.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/client/TokenRelayRequestInterceptor.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/config/FeignConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/client/AuthorizationHeaderUtilTest.java": {
    "stateCleared": "modified",
  },
}
`;
